// The BigQuery Storage API expects protocol buffer data to be encoded in the
// proto2 wire format. This allows it to disambiguate missing optional fields
// from default values without the need for wrapper types.

// Define a message type representing the rows in your table. The message
// cannot contain fields which are not present in the table.

// Use the required keyword for client-side validation of required fields.

syntax = "proto2";

message StockPrice {
    required string ticker = 1;
    required string market_time_utc = 2;
    required double open_price = 3;
    required double high_price = 4;
    required double low_price = 5;
    required double close_price = 6;
    optional double adj_close = 7;
    required int64  volume = 8;
    required string streamed_utc = 9;
}